src/header.jl

```diff
@@ -422,3 +422,33 @@ function show(io::IO, hdr::FITSHeader)
         i != n && println(io)
     end
 end
+
+"""
+    get_default_header(data::AbstractArray)
+
+Creates a default header from an `AbstractArray` with fields being `SIMPLE`, `BITPIX`, `NAXIS`, axis lengths and `EXTEND`.
+"""
+function get_default_header(data::AbstractArray{T}) where T <: Number
+    # assigning keys
+    hdu_keys = ["SIMPLE",
+                "BITPIX",
+                "NAXIS",
+                [string(Symbol("NAXIS", i)) for i in 1:ndims(data)]...,
```

Comment:
`["NAXIS$i" for i in 1:ndims(data)]...`

is there a way to accomplish this withou allocating this intermediate array? (does this even allocate the intermediate array?)

Reply:
There were around 60 allocations on this version (`@benchmark get_default_header(randn(512, 512))`), the newer version comes down to 48.